### DISCUSSION TITLE
Penetration axis magnitude

### DISCUSSION THREAD

**Discussion by mihe** [jrouwe/JoltPhysics#491] [Q&A] [answered]
Will `CollideShapeResult::mPenetrationAxis` always have a magnitude greater than 0, so that I can safely normalize it at all times?

The comment on `mPenetrationAxis` seems to imply that it will, by saying that I can just normalize it and invert it to get the contact normal, but what would the penetration axis be if the penetration depth is 0, which the comment on `mContactPointOn2` seems to imply is a possibility?

When looking around in the code I'm seeing some places do `mPenetrationAxis.NormalizedOr(Vec3::sZero())`, whereas other do `mPenetrationAxis.Normalized()`, so I'm not quite sure what to believe.

**Comment by jrouwe** [comment, marked as answer]
That's a difficult question :)

I think it should never return a zero length penetration axis, and all the simulation code just uses a Normalize() (I see I only used NormalizedOr(..) in the character). I've never seen it assert/crash on divisions by zero, but with floating point you never know.

**Reply by mihe** [reply]
Fair enough. :)

I figured I should at least check to see how worried I should be.

Thank you!
